src/wallet/wallet.cpp

```diff
@@ -1294,8 +1297,8 @@ bool CWallet::SetHDMasterKey(const CPubKey& pubkey)
 {
     LOCK(cs_wallet);
 
-    // ensure this wallet.dat can only be opened by clients supporting HD
-    SetMinVersion(FEATURE_HD);
+    // ensure this wallet.dat can only be opened by clients supporting HD with chain split
+    SetMinVersion(FEATURE_HD_SPLIT);
```

Comment:
This also forcibly upgrades the user's wallet (if from pre-split HD version) if they encrypt their wallet, which I think is unaccptable.